server/src/main/java/org/elasticsearch/action/support/tasks/BaseTasksRequest.java

```diff
@@ -52,9 +52,20 @@
 
     private TaskId taskId = TaskId.EMPTY_TASK_ID;
 
+    // NOTE: This constructor is only needed, because the setters in this class,
+    // otherwise it can be removed and above fields can be made final.
     public BaseTasksRequest() {
     }
 
+    protected BaseTasksRequest(StreamInput in) throws IOException {
```

Comment:
Can you move `writeTo` up under all of the reading ctors? It makes it much easier to eyeball that they match up when they are right next to each other.